includes/admin/class-wc-admin-notices.php

```diff
@@ -89,7 +89,7 @@ public static function reset_admin_notices() {
 			WC_Admin_Notices::add_notice( 'simplify_commerce' );
 		}
 
-		if ( ! is_ssl() || 'https' !== substr( $shop_page, 0, 5 ) ) {
+		if ( ! isset( $_SERVER['REQUEST_SCHEME'] ) && ( ! is_ssl() || 'https' !== substr( $shop_page, 0, 5 ) ) ) {
 			WC_Admin_Notices::add_notice( 'no_secure_connection' );
```

Comment:
Unless I'm missing something, I think this should probably check `isset ( $_SERVER['SOMETHING'] )` instead of `! isset( $_SERVER['SOMETHING'] )`.